server/sonar-server/src/main/java/org/sonar/server/sticker/ws/MeasureAction.java

```diff
@@ -19,54 +19,193 @@
+    String metricKey = request.mandatoryParam(PARAM_METRIC);
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      ComponentDto project = componentFinder.getByKeyAndOptionalBranch(dbSession, projectKey, branch);
+      userSession.checkComponentPermission(USER, project);
+      MetricDto metric = dbClient.metricDao().selectByKey(dbSession, metricKey);
+      checkState(metric != null && metric.isEnabled(), "Metric '%s' hasn't been found", metricKey);
+      LiveMeasureDto measure = getMeasure(dbSession, project, metricKey);
+      write(generateSvg(metric, measure), response.stream().output(), UTF_8);
+    } catch (StickerException | ForbiddenException | NotFoundException e) {
+      write(svgGenerator.generateError(e.getMessage()), response.stream().output(), UTF_8);
+    }
+  }
+
+  private LiveMeasureDto getMeasure(DbSession dbSession, ComponentDto project, String metricKey) {
+    return dbClient.liveMeasureDao().selectMeasure(dbSession, project.uuid(), metricKey)
+      .orElseThrow(() -> new StickerException(format("Measure '%s' has not been found for project '%s' and branch '%s'", metricKey, project.getKey(), project.getBranch())));
+  }
+
+  private String generateSvg(MetricDto metric, LiveMeasureDto measure) {
+    String metricType = metric.getValueType();
+    switch (ValueType.valueOf(metricType)) {
+      case INT:
+        return generateBadge(metric, formatNumber(getNonNullValue(measure, LiveMeasureDto::getValue).longValue()), Color.DEFAULT);
+      case PERCENT:
+        return generateBadge(metric, PERCENT_FORMATTER.format(getNonNullValue(measure, LiveMeasureDto::getValue)), Color.DEFAULT);
+      case LEVEL:
+        return generateQualityGate(metric, measure);
+      case WORK_DUR:
+        return generateBadge(metric, durations.format(Duration.create(getNonNullValue(measure, LiveMeasureDto::getValue).longValue())), Color.DEFAULT);
+      case RATING:
+        return generateRating(metric, measure);
+      default:
+        throw new IllegalStateException(format("Invalid metric type '%s'", metricType));
+    }
+  }
+
+  private String generateQualityGate(MetricDto metric, LiveMeasureDto measure) {
+    Level qualityGate = Level.valueOf(getNonNullValue(measure, LiveMeasureDto::getTextValue));
+    return generateBadge(metric, QUALITY_GATE_MESSAGE_BY_STATUS.get(qualityGate), COLOR_BY_QUALITY_GATE_STATUS.get(qualityGate));
+  }
+
+  private String generateRating(MetricDto metric, LiveMeasureDto measure) {
+    Rating rating = valueOf(getNonNullValue(measure, LiveMeasureDto::getValue).intValue());
+    return generateBadge(metric, rating.name(), COLOR_BY_RATING.get(rating));
+  }
+
+  private String generateBadge(MetricDto metric, String value, Color color) {
+    return svgGenerator.generateBadge(METRIC_NAME_BY_KEY.get(metric.getKey()), value, color);
+  }
+
+  private static <PARAM> PARAM getNonNullValue(LiveMeasureDto measure, Function<LiveMeasureDto, PARAM> function) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. ([squid:S00119](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00119))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10263%2Fmeasure_badge&issues=AWD-yF0lEFfjxkQcPxIs&open=AWD-yF0lEFfjxkQcPxIs)

[ISSUE_KEY]: AWD-yF0lEFfjxkQcPxIs

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED